cpp/src/arrow/compute/exec/asof_join_node_test.cc

```diff
@@ -92,6 +95,12 @@ Result<BatchesWithSchema> MakeBatchesFromNumString(
         ARROW_ASSIGN_OR_RAISE(Datum as_string, Cast(num_batch.values[i], utf8()));
         ARROW_ASSIGN_OR_RAISE(Datum as_type, Cast(as_string, type));
         values.push_back(as_type);
+      } else if (Type::BOOL == type->id()) {
+        ARROW_ASSIGN_OR_RAISE(Datum div_two, Divide(num_batch.values[i], two));
```

Comment:
What do we use these for?

Reply:
Oh it's to generate the final bool I see

Reply:
This extracts the low bit of the number parsed in `num_batch.values[i]`. If we denote the number as `x` then `as_bool` is computed as `(bool)(x - 2*(x/2))` which is the low bit cast to Boolean. It might not be the best way to compute this, but I didn't have time to find a better one. The low bit is useful here because it varies between `true` and `false`, so the tests obtain good coverage.